lib/rpush/daemon/apns2/delivery.rb

```diff
@@ -22,6 +22,15 @@ def perform
 
           # Send all preprocessed requests at once
           @client.join(timeout: CLIENT_JOIN_TIMEOUT)
+
+          # If there are notifications that are not yet delivered
+          # Usually means that there was a Connection reset error
+          # Mark remaining notifications for bounded retry
```

Comment:
If there are failed/successful notifications in the batch, they will be ignored.
Only unmarked notifications will be retried.